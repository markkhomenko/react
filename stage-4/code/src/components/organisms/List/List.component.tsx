import React from "react";
import * as Styled from "./List.styles";

interface ListProps {
  data: any[];
}

function List(props: ListProps): JSX.Element {
  const { data } = props;
  return (
    <Styled.Container>
      {data.map((repo: any) => (
        <li>
          <a href={repo.html_url}>{repo.name}</a>
        </li>
      ))}
    </Styled.Container>
  );
}
export default List as React.FC<ListProps>;
