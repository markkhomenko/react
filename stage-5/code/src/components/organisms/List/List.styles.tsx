import styled from "styled-components";

export const Container = styled.ul`
  width: 100%;
  max-width: 600px;
  li {
    a {
      font-family: "Open Sans", sans-serif;
    }
  }
`;
