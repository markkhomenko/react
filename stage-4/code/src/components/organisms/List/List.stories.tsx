import React from "react";

import { storiesOf } from "@storybook/react";

import List from "./List.component";

storiesOf("List", module).add("default", () => (
  <List data={["this", "is", "a", "list"]} />
));
