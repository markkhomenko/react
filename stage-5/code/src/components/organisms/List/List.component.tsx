import React from "react";
import * as Styled from "./List.styles";

import { Repo } from "../../../modules/api/github";

interface ListProps {
  data: Repo[];
}

function List(props: ListProps): JSX.Element {
  const { data } = props;
  return (
    <Styled.Container>
      {data.map((repo: Repo, index: number) => (
        <li key={index}>
          <a href={repo.html_url || ""}>{repo.name}</a>
        </li>
      ))}
    </Styled.Container>
  );
}
export default List as React.FC<ListProps>;
