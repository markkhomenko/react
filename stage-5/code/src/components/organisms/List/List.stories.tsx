import React from "react";

import { storiesOf } from "@storybook/react";

import { Repo } from "../../../modules/api/github";
import List from "./List.component";

const data = [
  { name: "this" },
  { name: "is" },
  { name: "a" },
  { name: "list" }
] as Repo[];

storiesOf("List", module).add("default", () => (
  <List data={data} />
));
