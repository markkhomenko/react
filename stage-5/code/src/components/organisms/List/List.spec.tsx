import React from "react";
import renderer from "react-test-renderer";

import { Repo } from "../../../modules/api/github";

import List from "./List.component";

test("List renders as expected", () => {
  const data = [
    { name: "this" },
    { name: "is" },
    { name: "a" },
    { name: "list" }
  ] as Repo[];

  const component = renderer.create(<List data={data} />);

  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
