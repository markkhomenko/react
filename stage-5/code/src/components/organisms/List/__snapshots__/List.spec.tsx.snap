// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List renders as expected 1`] = `
<ul
  className="sc-AxjAm juAXlR"
>
  <li>
    <a />
  </li>
  <li>
    <a />
  </li>
  <li>
    <a />
  </li>
  <li>
    <a />
  </li>
</ul>
`;
